sled-agent/src/rack_setup/service.rs

```diff
@@ -988,6 +1006,119 @@ impl ServiceInner {
         // DNS configuration to the internal DNS servers.
         self.initialize_dns(&service_plan).await?;
 
+        // Initialize rack network before NTP comes online, otherwise boundary
+        // services will not be available and NTP will fail to sync
+        info!(self.log, "Checking for Rack Network Configuration");
+        if let Some(rack_network_config) = &config.rack_network_config {
+            info!(self.log, "Initializing Rack Network");
+            info!(self.log, "Looking up address for Dendrite");
+            let resolver = DnsResolver::new_from_subnet(
+                self.log.new(o!("component" => "DnsResolver")),
+                config.az_subnet(),
+            )
+            .expect("Failed to create DNS resolver");
+
+            let dpd_addr = resolver
+                .lookup_socket_v6(internal_dns::ServiceName::Dendrite)
+                .await
+                .map_err(|e| SetupServiceError::DnsResolver(e))?;
+
+            let dpd = DpdClient::new(
+                &format!("http://[{}]:{}", dpd_addr.ip(), dpd_addr.port()),
+                dpd_client::ClientState {
+                    tag: "sled-agent".to_string(),
+                    log: self.log.new(o!(
+                        "component" => "DpdClient"
+                    )),
+                },
+            );
+
+            info!(self.log, "Building Rack Network Configuration");
+            let body = dpd_client::types::Ipv6Entry {
+                addr: BOUNDARY_SERVICES_ADDR.parse().map_err(|e| {
```

Comment:
Need a follow-up issue for dynamically assigning boundary services addresses as this will not work for multiple switches.

Reply:
https://github.com/oxidecomputer/omicron/issues/3278